### ISSUE TITLE
fine-tuning Few-Shot Relation Classification

### ISSUE THREAD

**Issue by hanjiale** [osainz59/Ask2Transformers#10]
Hi, 
Thank you for sharing this wonderful work! Could you detail the hyperparameters of fine-tuning TACRED?
I run the experiments with 

`python run_glue.py \
  --model_name_or_path roberta-large-mnli \
  --train_file /data/tacred/train.mnli.json \
  --validation_file /data/tacred/dev.mnli.json \
  --do_train \
  --do_eval \
  --max_seq_length 128 \
  --per_device_train_batch_size 32 \
  --learning_rate 4e-6 \
  --num_train_epochs 2 \
  --overwrite_output_dir \
  --fp16 True \
  --gradient_accumulation 1 \
  --output_dir ./results \
  --save_steps 5000 \
  --seed 0 \
  --warmup_steps 1000`

and then evaluate the final results with 
`python evaluation.py --config ../resources/predefined_configs/tacred.relation.config.json` (modify the "nli_models" with tuned model path)

the test resutls are
`"test": {"optimal_threshold": 0.5,
               "positive_accuracy": 0.9004511278195488,
               "precision": 0.43177373251090717,
               "recall": 0.8631578947368421,
               "f1-score": 0.5756117127958283
       }`

Could u kindly correct me how to tune and evaluate? Thank you!

**Comment by osainz59** [comment]
Hi @hanjiale ,

The hyperparameters you have used looks fine to me. Maybe I can suggest you to try 1e-5 as learning rate and increase the epochs to 5. You can also use the `load_best_model_at_end` with accuracy (or defining the entailment f1_score) as `metric_for_best_model` to improve the checkpoint selection. But in any case, those hyperparameters should be fine. I am concerned about the `optimal_threshold` value... Did you use the development set, at least on a small portion of it,  to optimize the threshold? Maybe that explains the huge gap between the precision and recall.

One of the weakness points of the approach is the need tof tunning the threshold, at least on a very small portion of data. We are working on it and have very promising results, but the improvements are still work in progress.

Let me know,
Oscar

**Comment by hanjiale** [comment]
I see. The problem is that I remove the `"dev_path"` in the `tacred.relation.config.json`. Thank you for your reply and looking forward to your new work!
